Imagine/Filter/PostProcessor/CwebpPostProcessor.php

```diff
@@ -0,0 +1,230 @@
+
+        unlink($file);
+
+        return $result;
+    }
+
+    protected function isBinaryTypeWebpImage(BinaryInterface $binary): bool
+    {
+        return $this->isBinaryTypeMatch($binary, ['image/webp']);
+    }
+
+    /**
+     * @param int|string[] $options
+     *
+     * @return string[]
+     */
+    private function getProcessArguments(array $options = []): array
+    {
+        $arguments = [$this->executablePath];
+
+        if ($nearLossless = $options['nearLossless'] ?? $this->nearLossless) {
+            if (!\in_array($nearLossless, range(0, 100), true)) {
+                throw new InvalidOptionException('The "nearLossless" option must be an int between 0 and 100', $options);
+            }
+
+            $arguments[] = '-near_lossless';
+            $arguments[] = $nearLossless;
+        }
+
+        if ($q = $options['q'] ?? $this->q) {
+            if (!\in_array($q, range(0, 100), true)) {
+                throw new InvalidOptionException('The "q" option must be an int between 0 and 100', $options);
+            }
+
+            $arguments[] = '-q';
+            $arguments[] = $q;
+        }
+
+        if ($alphaQ = $options['alphaQ'] ?? $this->alphaQ) {
+            if (!\in_array($alphaQ, range(0, 100), true)) {
+                throw new InvalidOptionException('The "alphaQ" option must be an int between 0 and 100', $options);
+            }
+
+            $arguments[] = '-alpha_q';
+            $arguments[] = $alphaQ;
+        }
+
+        if ($m = $options['m'] ?? $this->m) {
+            if (!\in_array($m, range(0, 6), true)) {
+                throw new InvalidOptionException('The "m" option must be an int between 0 and 6', $options);
+            }
+
+            $arguments[] = '-m';
+            $arguments[] = $m;
+        }
+
+        if ($alphaFilter = $options['alphaFilter'] ?? $this->alphaFilter) {
+            if (!\in_array($alphaFilter, ['none', 'fast', 'best'], true)) {
+                throw new InvalidOptionException('The "alphaFilter" option must be a string (none, fast or best)', $options);
+            }
+
+            $arguments[] = '-alpha_filter';
+            $arguments[] = $alphaFilter;
+        }
+
+        $alphaMethod = $options['alphaMethod'] ?? $this->alphaMethod;
+        if (null !== $alphaMethod) {
+            if (!\in_array($alphaMethod, range(0, 1), true)) {
+                throw new InvalidOptionException('The "alphaMethod" option must be an int between 0 and 1', $options);
+            }
+
+            $arguments[] = '-alpha_method';
+            $arguments[] = $alphaMethod;
+        }
+
+        if ($options['exact'] ?? $this->exact) {
+            $arguments[] = '-exact';
+        }
+
+        if ($metadata = $options['metadata'] ?? $this->metadata) {
+            $metadataValues = explode(',', $metadata);
+
+            foreach ($metadataValues as $metadataValue) {
+                if (!\in_array($metadataValue, ['all', 'none', 'exif', 'icc', 'xmp'], true)) {
```

Comment:
what does the combination of all or none with any other value mean? should we detect those as well, or do we leave that to cwebp to report?

Reply:
I don't know what to do about this. The `cwebp` documentation is not clear either: `Note that each input format may not support all combinations.`

Nothing is report, it just seems to be silently processed by the binary.

Reply:
thanks for checking. that is good enough then.